corehq/apps/locations/models.py

```diff
@@ -680,7 +680,11 @@ def save(self, *args, **kwargs):
         kwargs['sync_to_sql'] = False  # only sync here
         super(Location, self).save(*args, **kwargs)
         if sync_to_sql:
-            self._migration_do_sync()
+            try:
+                self._migration_do_sync()
+            except Exception:
+                self.delete()  # roll back on failure to sync
```

Comment:
hmm yeah that sounds tough. i wonder if might just be simpler to keep track of any failures and deal with them manually since there shouldn't be many